server/src/main/java/org/elasticsearch/common/cache/Cache.java

```diff
@@ -742,25 +733,29 @@ public long getEvictions() {
         }
     }
 
-    private boolean promote(Entry<K, V> entry, long now) {
-        boolean promoted = true;
+    private void promote(Entry<K, V> entry, long now) {
```

Comment:
Just a minor thought here, splitting this method to avoid double locking may not be worth the complexity. ReentrantLock will check if a lock is already owned by the same thread and avoid the compare and set operation. There's still going to be a volatile int field increment, but it should be relatively inexpensive.

Reply:
Fair point :) reverted that part of the change